collectors/osv/collectors.py

```diff
@@ -0,0 +1,275 @@
+                # CVE ID we can create a unique ID by appending that CVE ID to the OSV ID.
+                external_id = f"{osv_id}/{cve_id}"
+
+                _, created = Snippet.objects.update_or_create(
+                    source=Snippet.Source.OSV,
+                    external_id=external_id,
+                    defaults={"content": content},
+                )
+                if created:
+                    created += 1
+                else:
+                    updated += 1
+            return created, updated
+
+    @staticmethod
+    def extract_content(osv_vuln: dict) -> tuple[str, list, dict]:
+        """Extract data from an OSV vuln and normalize to Flaw model fields.
+
+        Fields that don't have their equivalents in the Flaw model can be used as well if we
+        think they may be useful in the future.
+        """
+        cve_ids = [
+            alias
+            for alias in osv_vuln.get("aliases", [])
+            if re.match(CVE_RE_STR, alias)
+        ]
+        osv_id = osv_vuln["id"]
+
+        def get_refs(data: dict) -> list:
+            #  https://ossf.github.io/osv-schema/#references-field
+            refs = [
+                {
+                    "type": FlawReference.FlawReferenceType.SOURCE,
+                    "url": f"https://osv.dev/vulnerability/{osv_id}",
+                }
+            ]
+            for ref in data.get("references", []):
+                refs.append(
+                    {
+                        "type": FlawReference.FlawReferenceType.EXTERNAL,
+                        "url": ref["url"],
+                    }
+                )
+            return refs
+
+        def get_description(data: dict) -> str:
+            #  https://ossf.github.io/osv-schema/#summary-details-fields
+            return data.get("details", "")
+
+        def get_title(data: dict) -> str:
+            #  https://ossf.github.io/osv-schema/#summary-details-fields
+            return data.get("summary", "")
+
+        def get_cvss(data: dict) -> list:
+            # https://ossf.github.io/osv-schema/#severity-field
+            scores = []
+            for score in data.get("severity", []):
+                if score["type"] == "CVSS_V3":
+                    cvss = CVSS3(score["score"])
+                    scores.append(
+                        {
+                            "issuer": FlawCVSS.CVSSIssuer.OSV,
+                            "score": float(cvss.base_score),
+                            "vector": cvss.vector,
+                            "version": FlawCVSS.CVSSVersion.VERSION3,
+                        }
+                    )
+                elif score["type"] == "CVSS_V2":
+                    cvss = CVSS2(score["score"])
+                    scores.append(
+                        {
+                            "issuer": FlawCVSS.CVSSIssuer.OSV,
+                            "score": float(cvss.base_score),
```

Comment:
This could be simplified as the `CVSS` model in `osidb` uses `cvss` lib. The model takes `vector` and than calculates a score.